src/datasets/summarizations/trend-weekly-comparison-average.summarization.ts

```diff
@@ -0,0 +1,41 @@
+
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  groupPointsByXWeek,
+} from './utils/time-series';
+import { formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {

```

Comment:
Add a description overview for this function

Reply:
Will copy the part of the description and sample summaries from the design doc in the refactor PR.